### PR TITLE
[PLT-7344] Revamp - Certification/Testing Page

### PR DESCRIPTION
**Pull request by amnambiar**
https://input-output.atlassian.net/browse/PLT-7344
https://input-output.atlassian.net/browse/PLT-7052

BE: [PR #90](https://github.com/input-output-hk/dapps-certification/pull/90) `origin/feat/l1-certificate` (also contains profile field modifications)

@coderabbitai: ignore

### Summary by CodeRabbit

Release Notes:

- New Feature: Introduced `Session`, `Landing`, `Certification`, and `TestingHistory` routes for enhanced navigation.
- New Feature: Added an `icon` prop to the `Button` component for improved customization.
- Refactor: Simplified HTML head section in `public/index.html` for better performance.
- Refactor: Updated `onRequestError` function in `src/api/onRequest.ts` to handle unauthorized requests more effectively.
- Refactor: Replaced `Loader` component with `CircularProgress` as the fallback for suspense in `App.tsx`.
- Refactor: Modified `CreateCertificate` component to use updated keys from `state.auth` object.
- Style: Enhanced visual appearance of primary button style and log view layout.
- Test: Prepared test cases for `LogsView` component (currently commented out).
- Chore: Updated color definitions and added new properties in `tailwind.config.js`.

### CODE REVIEW

**Comment by RSoulatIOHK** [comment]
## Full report
### Scrolling
When looking at the full report, there are 2 horizontal scroll bars. Can we have at most 1 or even better 0 and line wrap the text? 

There is a vertical scroll bar but it's kind of hidden on the far right and you need to use the right horizontal scroll bar to find it. Can we have it, on the right, at all times? I don't think people use the scroll bar to actually navigate but it will display that there is more information to see.

https://github.com/input-output-hk/dapps-certification-web/assets/117812549/50a6cfee-80c2-4db5-aa5e-8dd1563d4c8f

### Display more information
It is unclear on this screen that you can click on the properties to show more information
image
Maybe just an arrow pointing downwards, with still the possibility to click anywhere you want to open the details.

### "Purchase a certificate"
The "Purchase a Certificate (1 ADA)" should not be here for failed runs.
image

## Testing 

### Logs
The log opens all over everything.
Can we have it open in between the timeline and the PBT results?

### "Blank" page at the end of testing
At the end of testing, the timeline is removed and no testing results is displayed to the user:
image

Could we have instead this kind of nice visualization, without the possibility to open the properties to see more detail results?
image

### Testing time still here from a precedent run
The testing time is still displayed from a previous run:
image

### Test again
Pressing test again gives you this screen:
image

Not too sure what's going on, I think a new timeline has been fitted in the middle.
The testing information on the left is empty.
The previous "Full report" and "Download report" from the previous run should not be here anymore.

**Comment by amnambiar** [comment]
@RSoulatIOHK , Fixes are in place to re-review.

**Full Report View**
> Scrolling

FIXED

> Display more information - Maybe just an arrow pointing downwards, with still the possibility to click anywhere you want to open the details.

DONE ; added up/down clickable arrows to see the details, retaining click on row 

> The "Purchase a Certificate (1 ADA)" should not be here for failed runs.

FIXED ; This button will not be present if there any failures in unit tests or other tasks

**Testing View**
> Logs - Can we have it open in between the timeline and the PBT results?

DONE

> "Blank" page at the end of testing

FIXED ; the same table while the run in-progress will be shown

> Testing time still here from a precedent run

FIXED ; all other issues related to state retention are as well fixed, [PLT-7245](https://input-output.atlassian.net/browse/PLT-7425) as well

> Test again - bug

FIXED ; all states in _New test_ and _Test again_ are fixed

**Others**
- Fixed Button styles

**Open**
- Github Repo Access needs to be thoroughly tested; also [PLT-7458](https://input-output.atlassian.net/browse/PLT-7458)
- While Testing is running, the % progress in properties within the table are fine. But once it is finished, the % of progress of the last property is not correctly updated.
- Property Based Testing card needs to integrated correctly
- Integrate PR #8 
- Cannot see all tasks within Full Report table, due to [PLT-7614](https://input-output.atlassian.net/browse/PLT-7614)

File: src/pages/certification/Certification.tsx [hunk 1]
```diff
@@ -1,343 +1,158 @@
...
+                setFormData(data);
+                onCertificationFormSubmit({
+                    runId: lsUuid as string,
+                    commitHash: commitHash,
+                    repo: username + "/" + repoName,
+                });
+            }
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [lsFormData]);
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/Certification.tsx]: check for !runId ensures it is triggered only when there is no active run in place. Ignoring

File: src/pages/certification/Certification.tsx [hunk 2]
```diff
@@ -1,343 +1,158 @@
...
-      setTimelineConfig(config);
-    } catch (e) {
-      handleErrorScenario();
-      console.error('Failed:', e);
+    const resetStates = (ended: boolean = false) => {
+        setRunEnded(ended)
+        setRunId("")
+        setDisableForm(false)
+        setClickedFormSubmit(false)
     }
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/Certification.tsx:46]: ignoring

File: tailwind.config.js [hunk 3]
```diff
@@ -1,31 +1,76 @@
...
+      },
+      padding: {
+        20: "20px",
+        22: "22px"
+      },
+      margin: {
+        5: "5px",
+        10: "10px",
+        20: "20px"
+      },
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply tailwind.config.js:65]: ignoring

File: src/pages/certification/Certification.tsx [hunk 4]
```diff
@@ -1,343 +1,158 @@
...
+                        />
+                    ) : (
+                        <div className="w-full text-center text-xl text-neutral-300 font-medium pt-48">
+                            {clickedFormSubmit ? <Loader /> : <span>Fill the testing form</span>}
+                        </div>
+                    )}
+                </div>
             </div>
-          </Form>
         </div>
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/Certification.tsx]: ignoring

File: src/pages/certification/Certification.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/Certification.tsx]: ignoring

File: src/pages/certification/Certification.tsx [hunk 5]
```diff
@@ -1,343 +1,158 @@
...
+    const onCertificationFormSubmit = (data: {
+        runId: string;
+        commitHash: string;
+        repo: string;
+    }) => {
+        setRunId(data.runId);
+        setRunCommitHash(data.commitHash);
+        setRepo(data.repo)
+        setDisableForm(true);
     };
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/Certification.tsx]: ignoring

File: src/pages/certification/Certification.tsx [hunk 6]
```diff
@@ -1,343 +1,158 @@
...
-        dispatch(setUuid(runId));
-        setCertificationUuid(runId);
-        setCommitHash(commit);
-      } catch (e) {
-        handleErrorScenario();
-        console.error('Failed:', e);
-      }
+    // const { features: subscribedFeatures } = useAppSelector((state) => state.auth);
+    const [lsFormData] = useLocalStorage(LocalStorageKeys.certificationFormData, "");
+    const [lsUuid] = useLocalStorage(LocalStorageKeys.certificationUuid, "");
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/Certification.tsx:18]: ignoring

File: src/pages/certification/components/TimelineView/TimelineView.tsx [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+  const [runStatus, setRunStatus] = useState("");
+  const [runState, setRunState] = useState("");
+  const [refetchMin, setRefetchMin] = useState(5);
+  const [coverageFile, setCoverageFile] = useState("");
+  const [fetchRunStatus, setFetchRunStatus] = useState(false);
+  const [apiFetching, setApiFetching] = useState(false);
+  const [timelineConfig, setTimelineConfig] = useState(TIMELINE_CONFIG);
+  const [resultData, setResultData] = useState<any>({});
+  const [unitTestSuccess, setUnitTestSuccess] = useState(true); // assuming unit tests will pass
+  const [plannedTestingTasks, setPlannedTestingTasks] = useState<PlanObj[]>([]);
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/TimelineView/TimelineView.tsx]: ignoring

File: src/pages/certification/components/TimelineView/TimelineView.tsx [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+  const abortTestRun = () => {
+    confirm({ title: "", description: "Are sure you want to abort this run!" })
+      .then(async () => {
+        await dispatch(
+          deleteTestHistoryData({ url: "/run/" + uuid + "?delete=true" })
+        );
+        onAbort();
+      })
+      .catch(() => {});
+  };
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/TimelineView/TimelineView.tsx:71]: ignoring for now

File: src/pages/certification/components/TimelineView/TimelineView.tsx [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+        // setSubmitting(false);
+        clearPersistentStates();
+        setApiFetching(false);
+      }
+      setTimelineConfig(config);
+    } catch (e) {
+      // handleErrorScenario();
+      console.error("Failed:", e);
+      triggerFormReset();
+    }
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/TimelineView/TimelineView.tsx]: ignoring for now

File: src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx [hunk 10]
```diff
@@ -0,0 +1,326 @@
...
+  clearForm = false,
+  testAgain = false,
+  onSubmit,
+  onError,
+  loadingRunId
+}) => {
+  const form: any = useForm({
+    schema: auditorRunTestFormSchema,
+    mode: "all",
+  });
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx:56]: ignoring

File: src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx [hunk 11]
```diff
@@ -0,0 +1,326 @@
...
+          errorObj.response.statusText + " - " + errorObj.response.data + ". ";
+      }
+      errorMsg += "Please recheck and try again.";
+      setShowError(errorMsg);
+      const timeout = setTimeout(() => {
+        clearTimeout(timeout);
+        setShowError("");
+      }, 5000);
+    }
+  };
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx]: ok

File: src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx [hunk 12]
```diff
@@ -0,0 +1,326 @@
...
+            });
+          }
+        }
+      } catch (e) {
+        handleError(e);
+      }
+      setSubmitting(false);
+    };
+    triggerAPI();
+  };
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx]: ignoring for now

File: src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx [hunk 13]
```diff
@@ -0,0 +1,326 @@
...
+        const timeout = setTimeout(async () => {
+          clearTimeout(timeout);
+          await dispatch(verifyRepoAccess({ owner: username, repo: repoName }));
+          localStorage.removeItem(LocalStorageKeys.accessToken);
+        }, 0);
+      })();
+    }
+    // the enclosed snippet is to be triggered only once right when the component is rendered to check if the url contains code (to validate if it is a redirect from github)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx]: ignoring for now

File: src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx [hunk 14]
```diff
@@ -0,0 +1,326 @@
...
+        dispatch(clearAccessStatus())
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [repoUrlChanges]);
+
+  useEffect(() => {
+    showConfirmConnection && confirmConnectModal();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [showConfirmConnection]);
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx:266]: ignoring

File: src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx [hunk 15]
```diff
@@ -0,0 +1,326 @@
...
+            required={true}
+            disabled={submitting}
+            {...form.register("subject")}
+          />
+        </div>
+      </Form>
+
+      {showError ? <Toast message={showError} /> : null}
+    </div>
+  );
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx]: ignoring for now

File: src/pages/certification/components/TimelineView/TimelineView.tsx [hunk 16]
```diff
@@ -0,0 +1,316 @@
...
+  }, [runStatus, runState]);
+
+  useDelayedApi(
+    async () => {
+      setFetchRunStatus(false); // to clear timeout until api response
+      triggerFetchRunStatus();
+    },
+    refetchMin * TIMEOFFSET, // delay in milliseconds
+    fetchRunStatus // set to false to stop polling
+  );
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/TimelineView/TimelineView.tsx:196]: ignoring for now

File: src/pages/certification/components/TimelineView/TimelineView.tsx [hunk 17]
```diff
@@ -0,0 +1,316 @@
...
+  useEffect(() => {
+    // Run on unmount
+    return () => {
+      setFetchRunStatus(false); // to stop polling for run status
+    };
+  }, []);
+
+  if (!runStatus) {
+    return <Loader />
+  }
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/TimelineView/TimelineView.tsx]: ignoring

File: src/components/RepoAccessStatus/RepoAccessStatus.tsx [hunk 18]
```diff
@@ -0,0 +1,26 @@
+/* eslint-disable jsx-a11y/alt-text */
+import classNames from "classnames";
+import { ACCESS_STATUS } from "./repoAccessStatus.config";
+
+const RepoAccessStatus: React.FC<{
+  status: string;
+  statusText?: string;
+}> = ({ status, statusText }) => {
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/RepoAccessStatus/RepoAccessStatus.tsx]: ok; set to ""

File: src/components/RepoAccessStatus/RepoAccessStatus.tsx [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+      {statusText ? (
+        <span style={{ color: color }} className="ml-2">
+          {statusText}
+        </span>
+      ) : null}
+    </label>
+  );
+};
+
+export default RepoAccessStatus;
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/RepoAccessStatus/RepoAccessStatus.tsx]: ignoring for now

File: src/pages/certification/components/DownloadResult/DownloadResult.tsx [hunk 20]
```diff
@@ -0,0 +1,20 @@
+import { Button } from "@mui/material";
+import { exportObjectToJsonFile } from "utils/utils";
+import DownloadIcon from '@mui/icons-material/Download';
+
+const DownloadResult = (resultData: any) => {
+
+  const handleDownloadResultData = (resultData: any) => {
+    exportObjectToJsonFile(resultData, "Testing Report.json");
+  };
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/DownloadResult/DownloadResult.tsx:9]: ignoring for now

File: src/pages/certification/components/DownloadResult/DownloadResult.tsx [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+  return <Button
+      className="button"
+      onClick={(_) => handleDownloadResultData(resultData)}
+      variant="contained" size="medium"
+      startIcon={<DownloadIcon />}
+    >
+      Download Report
+    </Button>
+}
+export default DownloadResult;
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/DownloadResult/DownloadResult.tsx:20]: ignoring for now

File: src/components/RepoAccessStatus/repoAccessStatus.config.ts [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+      color: "#009168",
+    },
+    notAccessible: {
+      className: "image",
+      "data-testid": "notAccessible",
+      src: "images/failed.svg",
+      alt: "not accessible",
+      color: "#FF493D",
+    },
+  };
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/RepoAccessStatus/repoAccessStatus.config.ts:23]: ignoring for now

File: src/components/ProgressCard/CircularProgressBar.tsx [hunk 23]
```diff
@@ -0,0 +1,40 @@
...
+      </svg>
+
+      <span className="absolute top-1/2 left-0 progress-text  w-full text-center text-2xl font-extrabold">
+        {progress}%
+      </span>
+    </div>
+  );
+};
+
+export default CircularProgressBar;
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/ProgressCard/CircularProgressBar.tsx]: done

File: src/hooks/useLocalStorage.ts [hunk 24]
```diff
@@ -4,39 +4,97 @@ import { Dispatch, SetStateAction, useEffect, useState } from "react";
...
+      if (typeof parsedValue === 'string' && initialValue === null) {
+        // if the initialValue of key is supposed to be null, and the stored value is a string, 
+        // then it is wrongly set and need to be cleared
+        return initialValue;
+      } else if (parsedValue === null) { 
+        // in case of possible null after JSON.parse
+        return initialValue;
+      } else {
+        return parsedValue;
+      }
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/hooks/useLocalStorage.ts]: Proceeded with this as per review in another PR; ignoring this comment

File: src/hooks/useLocalStorage.ts [hunk 25]
```diff
@@ -52,21 +110,26 @@ function useLocalStorage<T>(
...
+          window.localStorage.setItem(key, serializedValue);
+
+          // Save state
+          setStoredValue(newValue);
+
+          // We dispatch a custom event so every useLocalStorage hook are notified
+          window.dispatchEvent(new Event("local-storage"));
+        } else {
+          removeValue()
+        }
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/hooks/useLocalStorage.ts]: done

File: src/pages/certification/Certification.helper.tsx [hunk 26]
```diff
@@ -84,4 +80,17 @@ export interface Log {
...
+}
+
+export const filterTaskKeysBy = (type: string) => {
+  return CertificationTasks.filter(item => item.type === type).map(item => item.key);
+}
+
+export const getCertificationTaskName = (key: string) => {
+  const obj: {key: string, label: string} | any = CertificationTasks.find(item => item.key === key)
+  return obj.label;
+};
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/Certification.helper.tsx:92]: ignoring

File: src/components/ProgressCard/ProgressCard.tsx [hunk 27]
```diff
@@ -0,0 +1,48 @@
...
+  totalValue: number;
+  displayText?: string;
+  color?: string;
+  tooltipText?: string
+}> = ({ color = "stroke-green", currentValue, totalValue, displayText, title, tooltipText }) => {
+  const [progress, setProgress] = useState(0)
+
+  useEffect(() => {
+    setProgress(Math.trunc((currentValue/totalValue) * 100))
+  }, [currentValue, totalValue])
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/ProgressCard/ProgressCard.tsx]: Done

[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/ProgressCard/ProgressCard.tsx]: Done

File: src/components/ProgressCard/ProgressCard.tsx [hunk 28]
```diff
@@ -0,0 +1,48 @@
...
+              {currentValue} / {totalValue}
+            </span>
+          }
+        </div>
+        <div className="mt-5 sm:mt-0 w-fit">
+          <CircularProgressBar progress={progress} color={color} />
+        </div>
+      </div>
+    );
+  }
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/ProgressCard/ProgressCard.tsx]: done

File: src/components/Table/components/ColViz/ColViz.tsx [hunk 29]
```diff
@@ -57,34 +67,44 @@ const ColViz: FC<any> = ({ columns, updateColumnOptions }) => {
...
+      }}
+    >
+      {columnDetails.filter((column: any) => column.Header.length).map((column: any, index: any) => (
+        <MenuItem key={index} onClick={() => !column.checkBoxDisabled ? handleChange(column.accessor) : null}>
+          <ListItemIcon>
+            <Checkbox checked={column.columnVisible} disabled={column.checkBoxDisabled} />
+          </ListItemIcon>
+          <ListItemText>{column.Header}</ListItemText>
+        </MenuItem>
+      ))}
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/Table/components/ColViz/ColViz.tsx]: done

File: src/components/Button/Button.tsx [hunk 30]
```diff
@@ -25,12 +26,13 @@ const Button: FC<IButtonProps> = ({
...
   onClick,
   ...props
 }) => {
   return (
     <button
-      className={`btn ${className} ${isLoading ? "is-loading" : ""} ${"btn-" + displayStyle} ${"btn-" + size}`}
+      className={`px-3 py-2 btn bg-secondary hover:bg-blue rounded-3 ${className} ${isLoading ? "is-loading" : ""} ${"btn-" + displayStyle} ${"btn-" + size}`}
       type={type}
       onClick={onClick}
       disabled={showLoader || disabled}
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/Button/Button.tsx:38]: ignoring

File: src/components/Button/Button.tsx [hunk 31]
```diff
@@ -25,12 +26,13 @@ const Button: FC<IButtonProps> = ({
...
   showLoader = false,
   displayStyle = "primary",
+  icon = null,
   onClick,
   ...props
 }) => {
   return (
     <button
-      className={`btn ${className} ${isLoading ? "is-loading" : ""} ${"btn-" + displayStyle} ${"btn-" + size}`}
+      className={`px-3 py-2 btn bg-secondary hover:bg-blue rounded-3 ${className} ${isLoading ? "is-loading" : ""} ${"btn-" + displayStyle} ${"btn-" + size}`}
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/Button/Button.tsx:35]: ignoring

File: src/hooks/useLogs.ts [hunk 32]
```diff
@@ -125,7 +120,7 @@ export const useLogs = (
                 dispatch(setBuildInfo())
             }
         } catch(e) {
-            handleErrorScenario();
+            handleErrorScenario && handleErrorScenario();
             console.log(e);
         } finally{
             setFetchingLogs(false)
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/hooks/useLogs.ts:126]: ignoring

File: src/components/Table/components/ColViz/ColViz.tsx [hunk 33]
```diff
@@ -17,17 +27,17 @@ const ColViz: FC<any> = ({ columns, updateColumnOptions }) => {
...
     const updatedColumns = columnDetails.map((column: any) => {
       if (column.accessor === id) {
-        column.columnVisible = e.target.checked;
+        column.columnVisible = !column.columnVisible;
       }
       return column;
     });
     atLeastOneItemChecked(updatedColumns);
     // setColumnDetails(updatedColumns);
   };
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/Table/components/ColViz/ColViz.tsx]: done

File: src/api/onRequest.ts [hunk 34]
```diff
@@ -17,7 +17,10 @@ export function onRequest(config: AxiosRequestConfig) {
 }
 
 export function onRequestError(error: AxiosError) {
-  // Do something with request error
+  if (localStorage.getItem(LocalStorageKeys.isLoggedIn) !== "true") {
+    localStorage.clear();
+    window.location.reload();
+  }
   return Promise.reject(error);
 }
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/api/onRequest.ts]: ignoring

File: tailwind.config.js [hunk 35]
```diff
@@ -1,34 +1,76 @@
...
+        xs: "375px" // => @media (min-width: 375px) { ... }
+      }
     },
   },
   corePlugins: {
     preflight: false,
   },
-  plugins: [],
+  plugins: []
 };
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply tailwind.config.js]: done

File: src/pages/certification/certification-result/FullReportTable.tsx [hunk 36]
```diff
@@ -0,0 +1,69 @@
...
+import './fullReportTable.css';
+
+const FullReportTable: React.FC<{
+  data: { [x: string]: any },
+}> = ({
+  data,
+}) => {
+
+    const tableData = processData(data)
+
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/certification-result/FullReportTable.tsx]: done

File: src/components/LogsView/LogsView.scss [hunk 37]
```diff
@@ -20,12 +42,14 @@
...
-        h6 {
+        h5 {
             float: left;
+            margin: 0;
         }
         .minimize-btn {
             cursor: pointer;
             font-size: xxx-large;
+            float: right;
         }
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/LogsView/LogsView.scss:41]: ignoring

File: src/pages/certification/certification-result/FullReportTable.tsx [hunk 38]
```diff
@@ -0,0 +1,69 @@
...
+            return <img
+              className="image"
+              src="/images/failed.svg"
+              alt="failure"
+            />
+          } else { return null; }
+        }
+      }
+    ],
+      [])
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/certification-result/FullReportTable.tsx]: done

File: src/pages/certification/certification-result/FullReportTable.tsx [hunk 39]
```diff
@@ -0,0 +1,69 @@
...
+              alt="success"
+            />
+          } else if (props.row.original.status === 'failure') {
+            return <img
+              className="image"
+              src="/images/failed.svg"
+              alt="failure"
+            />
+          } else { return null; }
+        }
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/certification-result/FullReportTable.tsx]: ignoring

File: src/pages/certification/components/TimelineView/TimelineView.tsx [hunk 40]
```diff
@@ -0,0 +1,306 @@
...
+  const [plannedTestingTasks, setPlannedTestingTasks] = useState<PlanObj[]>([]);
+
+  const abortTestRun = () => {
+    confirm({ title: "", description: "Are sure you want to abort this run!" })
+      .then(async () => {
+        await dispatch(
+          deleteTestHistoryData({ url: "/run/" + uuid + "?delete=true" })
+        );
+        onAbort();
+      })
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/TimelineView/TimelineView.tsx:69]: ignoring for now

File: src/components/LogsView/LogsView.tsx [hunk 41]
```diff
@@ -1,45 +1,58 @@
...
+    const { logInfo: logs } = useLogs(
+        runId,
+        oneTime || endPolling,
+        !oneTime
+    )
+
+
+    useEffect(() => {
+        bottomRef.current?.scrollIntoView({ behavior: 'smooth' }); // scroll to bottom 
+    }, [logs])
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/LogsView/LogsView.tsx:34]: ignoring

File: src/components/LogsView/LogsView.tsx [hunk 42]
```diff
@@ -54,15 +67,15 @@ const InformationTable: FC<{logs: any, }> = ({ logs, }) => {
...
                                     logData = attr['fields']['chunk-data']
                                 }
-                            } catch(e) {
+                            } catch (e) {
                                 // do nothing
                                 if (typeof item.Text == 'string' && item.Text.length) {
                                     logData = item.Text
                                 }
                             }
                             logData = !logData.length ? item.Text : logData
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/LogsView/LogsView.tsx:76]: ignoring for now

File: src/pages/certification/components/FileCoverageContainer.tsx [hunk 43]
```diff
@@ -75,39 +76,43 @@ const FileCoverageContainer: React.FC<{
...
-        </div>
-    );
+    return (<>
+        {coverageIndexFiles ? (
+            <div id="coverageIndicator">
+                <ul>{renderRows()}</ul>
+            </div>
+        ) : null}
+    </>);
 };
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/FileCoverageContainer.tsx]: ignoring for now

File: src/pages/certification/components/TimelineView/TimelineView.tsx [hunk 44]
```diff
@@ -0,0 +1,306 @@
...
+        // setSubmitting(false);
+        clearPersistentStates();
+        setApiFetching(false);
+      }
+      setTimelineConfig(config);
+    } catch (e) {
+      // handleErrorScenario();
+      console.error("Failed:", e);
+      triggerFormReset();
+    }
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/TimelineView/TimelineView.tsx]: ignoring for now

File: src/pages/testingHistory/index.tsx [hunk 45]
```diff
@@ -233,10 +233,18 @@ const TestHistory = () => {
...
+  }
+
   const columns = React.useMemo(() => [
     {
       Header: "Repo name",
       accessor: "repoUrl",
+      Cell: (props: any) => (
+        <span>{formatRepoUrl(props.row.original.repoUrl)}</span>
+      )
     },
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/testingHistory/index.tsx:248]: ignoring for now

File: src/pages/certification/Certification.tsx [hunk 46]
```diff
@@ -1,343 +1,155 @@
...
+                setData(formData);
+                onCertificationFormSubmit({
+                    runId: uuid as string,
+                    commitHash: commitHash,
+                    repo: username + "/" + repoName,
+                });
+            }
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [lsFormData]);
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/Certification.tsx]: ignoring for now

File: src/pages/certification/components/AuditorRunTestForm/auditorRunTestForm.schema.ts [hunk 47]
```diff
@@ -0,0 +1,76 @@
...
+              message: "Invalid GitHub repository URL",
+            });
+          }
+
+          return true;
+        } else {
+          return true;
+        }
+      },
+    }),
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/components/AuditorRunTestForm/auditorRunTestForm.schema.ts]: done

**Comment by RSoulatIOHK** [comment]
Still have the double scrolling bars and the vertical one is far away to see.
image

When the repo doesn't exist, I get this Uncaught error
image

Tool tip and check for repo accessibility are over one another:
image

I know it's in the open issues but the progress % are not fine for most of the properties on some repo testing:
image
But is correct on the full report:
image

I still see the previous testing time:
image

Also the Github repo check is a bit agressive. I haven't managed to type the repo myself without triggering the check, could we just have it when you're no longer in the input bar?

All the rest looks fixed :)

**Comment by amnambiar** [comment]
@RSoulatIOHK , Thank you for the patience :bowing_woman: 

> Still have the double scrolling bars and the vertical one is far away to see.

Fixed for good now. Pardon :disappointed: 
Reason - mui-Container element had `maxWidth`defaulted to `'lg'`. I had changed it to `'xl'` instead of setting it `false`. Is set to `false` now.

> - When the repo doesn't exist, I get this Uncaught error
> - Tool tip and check for repo accessibility are over one another:

Fixed

> I still see the previous testing time:

Fixed :sweat_smile: 

>  Github repo check - could we just have it when you're no longer in the input bar

Triggering the check on blur breaks at multiple scenarios, hence retained the check on type itself but with a lesser timeout than the older version of tool. It is more quicker, doesn't let [PLT-7458 ](https://input-output.atlassian.net/browse/PLT-7458) happen (Test btn will be disabled until the repo is accessible).

> progress % issue

Raised [PLT-7646](https://input-output.atlassian.net/browse/PLT-7646) to track this

**Comment by RSoulatIOHK** [comment]
I still have the scroll bars :( If I put the window on my 4k screen, i don't have them but that might be a strong requirement for our users 😄 
image
(Let's put this one on the Tech debt if you want)

Rest is fine :) (apart from the tickets you have raised to track some of the bigger issues)

**Comment by amnambiar** [comment]
@RSoulatIOHK ,

> Multiple scrollbars above 4k 

I've added a fix for the multiple scrollbars. Hope it is resolved for you now. Anyhow I've raised a ticket [PLT-7656](https://input-output.atlassian.net/browse/PLT-7656). We can track it there.

Thanks!

**Review by RSoulatIOHK — APPROVED** [review]
Looks good!

**Comment by amnambiar** [comment]
Rechecked 'Testing' the form, timeline view, full report, state retention and it looks fine.

**Comment by RSoulatIOHK** [comment]
Apart from the issues in the timeline results when it ends (sometimes)

image

The rest looks good!

**Pull request merged**